### PR TITLE
Fix unconfirmed balance, accurately report fees

### PR DESCRIPTION
**Pull request by AwfulCrawler**
Wallet balance can be misreported when there are unconfirmed transactions with dust amounts generated from change.

If dust is meant to be added to fee or sent elsewhere to a specific dust address, the change amount was not updated to reflect this.

Suggested fix:

I have mainly modified transfer, transfer_selected.  After tx details have been worked out, do the following:
   -ptx.fee has dust explicitly added to it if specified by the dust policy.
   -change_dts.amount has dust subtracted from it if dust is added to fee or sent elsewhere 
   -dust is zero if it is not added to fee and not sent elsewhere (i.e. it's still in the change)

In addition I've made the confirm prompt in simple_wallet::transfer a little bit more verbose so the user knows more about what's going on in regards to dust amounts.

I have not looked at sweep_unmixable at all.

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -2202,24 +2202,36 @@ bool simple_wallet::transfer_main(bool new_algorithm, const std::vector<std::str
...
         {
-          prompt_str = (boost::format(tr("The transaction fee is %s.  Is this okay?  (Y/Yes/N/No)")) %
-            print_money(total_fee)).str();
+          prompt << boost::format(tr("The transaction fee is %s")) %
+            print_money(total_fee);
         }
-        std::string accepted = command_line::input_line(prompt_str);
+        if (dust_in_fee != 0) prompt << boost::format(tr(", of which %s is dust from change")) % print_money(dust_in_fee);
+        if (dust_not_in_fee != 0)  prompt << tr(".") << ENDL << boost::format(tr("A total of %s from dust change will be sent to dust address")) 
+                                                   % print_money(dust_not_in_fee);
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: Why does this need to be handled separately since AFAIK it doesn't go to a different address ?

**Comment by moneromooo-monero** [comment]
Seems ok. Not sure we want to keep the dust field now though. It's all a bit confusing to me.

File: src/simplewallet/simplewallet.cpp (hunk 1 above)
Comment (AwfulCrawler) [inline src/simplewallet/simplewallet.cpp]: I guess it doesn't need to be handled separately at the moment.  

It's just that addr_for_dust exists as a field at the moment in tx_dust_policy, which allows for the possibility of a dust address in the future.  It seems that this field is ignored at the moment though.

My preference would probably be to remove dust and dust policy and just have everything go back as change, which would fix the bug and (as per your comment below) make things less confusing.  However I'm a bit new to the code so I thought there might be a good reason why the dust handling was there in the first place.

Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: There used to be a dust limit, and amounts below this were not decomposed. The decomposition to the last decimal place is a new thing. I guess I should take some time to see whether removing that dust field would have any side effect.

**Pull request merged**